app/src/main/java/com/example/sharewhatyoucanproject/donor/DonorActivity.kt

```diff
@@ -0,0 +1,162 @@
+package com.example.sharewhatyoucanproject
+
+import android.Manifest
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.provider.MediaStore
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.databinding.ActivityDonorBinding
+import com.example.sharewhatyoucanproject.utils.PERMISSION_ALL
+import com.example.sharewhatyoucanproject.utils.foodTypes
+import com.example.sharewhatyoucanproject.utils.hasPermissions
+import com.example.sharewhatyoucanproject.utils.isGPSEnabled
+import com.example.sharewhatyoucanproject.utils.locationPermissions
+import com.example.sharewhatyoucanproject.utils.showToast
+import com.google.android.gms.location.LocationRequest
+import com.google.android.material.progressindicator.CircularProgressIndicator
+
+class DonorActivity : AppCompatActivity() {
+
+    lateinit var donorViewModel: DonorViewModel
+    private lateinit var binding: ActivityDonorBinding
+    lateinit var locationRequest: LocationRequest
+    lateinit var arrayAdapter: ArrayAdapter<String>
+
+    lateinit var circularProgressIndicator: CircularProgressIndicator
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityDonorBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        circularProgressIndicator = binding.progressCircular
+        locationRequest = LocationRequest.create()
+        supportActionBar?.hide()
+        donorViewModel = ViewModelProvider(
+            this,
+            DonorViewModelFactory(application),
+        )[DonorViewModel::class.java]
+
+        donorViewModel.currentLocation.observe(this) {
+            it?.let {
+                donorViewModel.uploadImage()
+            }
+        }
+        donorViewModel.imageUrl.observe(this) {
+            it?.let { donorViewModel.saveData() }
+        }
+
+        donorViewModel.donorResult.observe(this) { donorResult ->
+            when (donorResult) {
+                is DonorResult.showMessage -> {
+                    if (donorResult.message == "Success") {
```

Comment:
It doesn't look like you need a `message` attribute for `Success` state:

```
sealed class DonorResult {
    object Success : DonorResult()
    Error(val message: String): DonorResult()
}
```
usage:
```
when (donorResult) {
  DonorResult.SUCCESS -> {
     // success logic 
  }
  is DonorResult.Error -> {
    //error logic
    showToast(donorResult.message)
  }
}
```